src/tools/illink/src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -816,11 +816,15 @@ bool RequiresInterfaceRecursively (TypeDefinition typeToExamine, TypeDefinition
 			return false;
 		}
 
-		void ProcessDefaultImplementation (TypeDefinition typeWithDefaultImplementedInterfaceMethod, InterfaceImplementation implementation)
+		void ProcessDefaultImplementation (TypeDefinition typeWithDefaultImplementedInterfaceMethod, InterfaceImplementation implementation, MethodDefinition implementationMethod)
 		{
-			if (!Annotations.IsInstantiated (typeWithDefaultImplementedInterfaceMethod))
+			if ((!implementationMethod.IsStatic && !Annotations.IsInstantiated (typeWithDefaultImplementedInterfaceMethod))
+				|| implementationMethod.IsStatic && !Annotations.IsRelevantToVariantCasting(typeWithDefaultImplementedInterfaceMethod))
 				return;
 
+			var origin = ScopeStack.CurrentScope.Origin;
+			MarkMethod(implementationMethod, new DependencyInfo(DependencyKind.Unspecified, implementation), in origin);
```

Comment:
It's a little different for instance DIMs because the check is "isInstantiated" vs "IsRelevantToVariantCasting" for static interfaces. If the most derived interface (e.g. IDerived) is instantiated, all the implemented interfaces are also marked instantiated, so the _interfaceOverrides checks will say that IMiddle.get_Value is needed because IMiddle is instantiated. However, for RelevantToVariantCasting, only the type is marked "RelevantToVariantCasting" so when it reaches the _interfaceOverrides checks, the type that provides the DIM (IMiddle) isn't RelevantToVariantCasting, even though the type (UsedAsIBase) that implements the interface with the static method base (IBase.get_Value) is marked RelevantToVariantCasting. We'd to check for "IsRelevantToVariantCasting" on the types that need the DIM as well.

I'm thinking maybe it's worth a try at unifying static and instance interface marking to see if I can simplify this.